public/app/features/dashboard-scene/settings/variables/editors/QueryVariableEditor.tsx

```diff
@@ -50,7 +50,15 @@ export function QueryVariableEditor({ variable, onRunQuery }: QueryVariableEdito
     variable.setState({ datasource });
   };
   const onQueryChange = (query: VariableQueryType) => {
-    variable.setState({ query });
+    let definition: string;
+    if (typeof query === 'string') {
+      definition = query;
+    } else if (query.hasOwnProperty('query') && typeof query.query === 'string') {
+      definition = query.query;
+    } else {
+      definition = '';
+    }
+    variable.setState({ query, definition });
```

Comment:
@ivanortegaalba just updating to string was not enough, so I found additional logic in `public/app/features/dashboard-scene/settings/variables/editors/QueryVariableEditor.tsx` `onQueryChange` function and added it here as well